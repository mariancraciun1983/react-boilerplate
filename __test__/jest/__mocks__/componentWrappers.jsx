import React  from 'react';
import clone from 'clone-deep';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import mockStore  from './store';


const { fullStore } = mockStore;

const genericWrapper = (children) => {
  const mockStore = configureStore();
  const initialState = clone(fullStore);
  const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  )
};

export {
  genericWrapper
};
