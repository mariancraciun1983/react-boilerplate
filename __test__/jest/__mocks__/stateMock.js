export default {
  app: {
    state: "loaded"
  },
  auth: {
    authenticated: true,
    token: "abc123",
    user: {
      name: "John Doe",
      email: "john@example.com",
      avatar: "https://dummyimage.com/100x100.png/5fa2d1/ffffff"
    }
  },
  genres: {
    list: [
      {
        name: "Comedy",
        slug: "comedy"
      },
      {
        name: "Adventure",
        slug: "adventure"
      }
    ]
  },
  movies: {
    list: [
      {
        id: "0b7b8b05-6c88-4517-a969-adf89a46bc0a",
        title: "Lost Palace",
        genre: ["comedy"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "Donec diam neque, vestibulum eget, vulputate ut, ultrices vel, augue. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Donec pharetra, magna vestibulum aliquet ultrices, erat tortor sollicitudin mi, sit amet lobortis sapien sapien non mi. Integer ac neque.\n\nDuis bibendum. Morbi non quam nec dui luctus rutrum. Nulla tellus.\n\nIn sagittis dui vel nisl. Duis ac nibh. Fusce lacus purus, aliquet at, feugiat non, pretium quis, lectus.\n\nSuspendisse potenti. In eleifend quam a odio. In hac habitasse platea dictumst.\n\nMaecenas ut massa quis augue luctus tincidunt. Nulla mollis molestie lorem. Quisque ut erat.",
        price: 2.39,
        slug: "gunner-palace",
        title_lower: "lost palace"
      },
      {
        id: "f01187b5-6fe8-465b-a0b8-3764a94ae225",
        title: "Love is Eternal While It Lasts",
        genre: ["comedy", "romance"],
        image: "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        description:
          "Curabitur gravida nisi at nibh. In hac habitasse platea dictumst. Aliquam augue quam, sollicitudin vitae, consectetuer eget, rutrum at, lorem.\n\nInteger tincidunt ante vel ipsum. Praesent blandit lacinia erat. Vestibulum sed magna at nunc commodo placerat.\n\nPraesent blandit. Nam nulla. Integer pede justo, lacinia eget, tincidunt eget, tempus vel, pede.",
        price: 9.5,
        slug: "love-is-eternal-while-it-lasts",
        title_lower: "love is eternal while it lasts"
      },
      {
        id: "23ad0474-8b8d-420d-a2cc-dd54aa6ed589",
        title: "Lost World, The",
        genre: ["romance"],
        image: "http://dummyimage.com/100x100.png/cc0000/ffffff",
        description:
          "In congue. Etiam justo. Etiam pretium iaculis justo.\n\nIn hac habitasse platea dictumst. Etiam faucibus cursus urna. Ut tellus.\n\nNulla ut erat id mauris vulputate elementum. Nullam varius. Nulla facilisi.\n\nCras non velit nec nisi vulputate nonummy. Maecenas tincidunt lacus at velit. Vivamus vel nulla eget eros elementum pellentesque.\n\nQuisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.",
        price: 7.4,
        slug: "lost-world-the",
        title_lower: "lost world, the"
      }
    ]
  },
  cart: {
    dict: {
      "0b7b8b05-6c88-4517-a969-adf89a46bc0a": 1,
      "23ad0474-8b8d-420d-a2cc-dd54aa6ed589": 2
    },
    saved: false
  },
  filters: {
    string: "Lo",
    string_lower: "lo"
  },
  ui: {
    language: "en",
    version: 2,
    theme: "light"
  },
  routing: {
    locationBeforeTransitions: {
      pathname: "/genre/comedy",
      search: "",
      hash: "",
      key: "sb1up4"
    }
  },
  form: {}
};
