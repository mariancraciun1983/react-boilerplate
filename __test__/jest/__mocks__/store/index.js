import clone from 'clone-deep';

import * as FullData from '../../../../__mock__/api';

const defaultStore = {
  app: {
    state: 'loading',
  },
  auth: {
    authenticated: false,
    token: null,
    user: null,
  },
  cart: {
    dict: {},
    saved: true,
  },
  filters: {
    string: '',
    string_lower: '',
  },
  genres: {
    list: null,
  },
  movies: {
    list: null,
  },
  ui: {
    language: 'en',
    version: 0,
    theme: 'light',
  },
  routing: {},
  form: {},
};

const fullStore = clone(defaultStore);
fullStore.state = 'loaded';
fullStore.genres.list = FullData.genres;
fullStore.movies.list = FullData.movies.slice(0,200);
export default {
  defaultStore,
  fullStore,
}
