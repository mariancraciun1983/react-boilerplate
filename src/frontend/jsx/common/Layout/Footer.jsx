import React from 'react';

class Footer extends React.PureComponent {
  render() {
    return (
      <footer className="border-top">
        <div className="text-center p-3">
          No Rights Reserved.
        </div>
      </footer>
    );
  }
}

export const component = Footer;
export default Footer;
