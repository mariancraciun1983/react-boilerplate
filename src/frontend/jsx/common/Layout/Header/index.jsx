import React from 'react';
import classnames from 'classnames';
import { Link } from 'react-router-dom';

import Tools from './Tools';
import Menu from './Menu';
import Search from './Search';

class Header extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = { toggled: false };
    this.toggle = this.toggle.bind(this);
  }

  toggle() {
    this.setState((state) =>{
      const { toggled } = state;
      return { toggled: !toggled };
    });
  }

  render() {
    const { toggled } = this.state;
    const classes = classnames({ 'navbar-collapse': true, collapse: !toggled });
    return (
      <>
        <nav className="navbar navbar-expand-lg mb-3">
          <Link to="/" className="navbar-brand">
            ReactJS
          </Link>
          <button className="navbar-toggler" type="button" onClick={this.toggle}>
            <span className="navbar-toggler-icon" />
          </button>
          <Search />
          <div className={classes}>
            <div className="mr-auto mt-2 mt-lg-0">
              <Tools />
            </div>
            <Menu />
          </div>
        </nav>
      </>
    );
  }
}

export const Component = Header;
export default Header;
