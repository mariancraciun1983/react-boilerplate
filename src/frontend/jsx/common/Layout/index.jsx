import React from 'react';
import PropTypes from 'prop-types';

import Header from './Header';
import Footer from './Footer';

class Layout extends React.PureComponent {
  render() {
    const { children } = this.props;
    return (
      <>
        <div className="holder">
          <Header />
          <div className="container-fluid">
            {children}
          </div>
        </div>
        <Footer />
      </>
    );
  }
}

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export const Component = Layout;
export default Layout;
