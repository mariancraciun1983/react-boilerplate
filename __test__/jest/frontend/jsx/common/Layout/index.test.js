/// <reference types="jest" />
import React, {Fragment} from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { genericWrapper } from '../../../../__mocks__/componentWrappers';

import { Component } from '../../../../../../src/frontend/jsx/common/Layout';

describe('components/common', () => {
  it('Shows the Layout', () => {
    expect.assertions(2);
    const wrapper = shallow(<Component><Fragment /></Component>);
    expect(wrapper.find('Header')).toHaveLength(1);
    expect(wrapper.find('Footer')).toHaveLength(1);
  });


  it('Renders the Header correctly', () => {
    expect.assertions(1);
    const tree = renderer
    .create(genericWrapper(<Component><Fragment /></Component>))
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

});
