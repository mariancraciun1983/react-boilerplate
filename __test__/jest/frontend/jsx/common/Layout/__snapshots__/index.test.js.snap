// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common Renders the Header correctly 1`] = `
Array [
  <div
    className="holder"
  >
    <nav
      className="navbar navbar-expand-lg mb-3"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        ReactJS
      </a>
      <button
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="form-inline my-2 my-lg-0"
      >
        <input
          className="form-control mr-sm-2"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
      <div
        className="navbar-collapse collapse"
      >
        <div
          className="mr-auto mt-2 mt-lg-0"
        >
          <nav
            className="d-flex flex-column flex-md-row float-right"
          >
            <button
              className="btn mr-2 btn-outline-light"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-paint-roller"
              />
              Dark Version
            </button>
            <button
              className="btn mr-2 btn-danger"
              onClick={[Function]}
              type="button"
            >
              <i
                className="fas fa-globe-americas"
              />
              ES
            </button>
          </nav>
        </div>
        <nav
          className="mr-5"
        >
          <a
            className="btn btn-warning"
            href="/cart"
            onClick={[Function]}
          >
            <i
              className="fas fa-shopping-cart"
            />
            Cart
            
          </a>
        </nav>
        <div
          className="btn-group"
          role="group"
        >
          <a
            className="btn btn-secondary"
            href="/auth/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            className="btn btn-primary"
            href="/auth/register"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </nav>
    <div
      className="container-fluid"
    />
  </div>,
  <footer
    className="border-top"
  >
    <div
      className="text-center p-3"
    >
      No Rights Reserved.
    </div>
  </footer>,
]
`;
